src/orquestra/sdk/_base/_api/_wf_run.py

```diff
@@ -431,11 +433,73 @@ def get_logs(self) -> WorkflowLogs:
         """
         return self._runtime.get_workflow_logs(wf_run_id=self.run_id)
 
-    # TODO: ORQSDK-617 add filtering ability for the users
-    def get_tasks(self) -> t.Set[TaskRun]:
-        wf_run_model = self.get_status_model()
+    def get_tasks(
+        self,
+        state: t.Optional[t.Union[State, t.List[State]]] = None,
+        function_name: t.Optional[str] = None,
+        task_run_id: t.Optional[t.Union[str, TaskRunId]] = None,
+        task_invocation_id: t.Optional[t.Union[str, ir.TaskInvocationId]] = None,
+    ) -> t.Set[TaskRun]:
+        """
+        Returns TaskRun representations of the tasks executed as part of this workflow.
 
-        return {
+        Args:
+            state: If specified, only tasks with matching states will be returned.
+            function_name: A function name, or regex string matching the desired
+                function name(s). If specified, only tasks with matching function names
+                will be returned.
+            task_run_id: A task run ID, or regex string matching the desired task run
+                ID(s). If specified, only tasks with matching task run IDs will be
+                returned.
+            task_invocation_id: A task invocation ID, or regex string matching the
+                desired task invocation ID(s). If specified, only tasks with matching
+                task invocation IDs will be returned.
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
**praise:** ‏This is nice!